client/my-sites/sidebar/index.jsx

```diff
@@ -851,9 +851,14 @@ export class MySitesSidebar extends Component {
 
 function mapStateToProps( state ) {
 	const currentUser = getCurrentUser( state );
-	const selectedSiteId = getSelectedSiteId( state );
+
+	const isAllDomainsView = isUnderDomainManagementAll( getCurrentRoute( state ) );
```

Comment:
This `MySitesSidebar` component is a generic one that can be used (potentially or really) in many different Calypso environments, like Jetpack Cloud or a management app for WP for Teams (aka P2).

But with this code, it "needs to know" about a very specific part of the Calypso app, namely the domain management section. Generic components shouldn't have knowledge about or dependencies on specific ones.

I think it's better to add support for a new prop -- `forceAllSitesView`, or `observeSelectedSite={ false }` -- and pass the value from above. That would be the code that wires the app view together, e.g., the `/me/domains` route handler.

Reply:
Cool - that's why I wanted someone with more experience to look into this part of the code. I'll update the PR and fix it! 🙇

Reply:
OKay - I looked into this but it seems that `MySitesSidebar` is not exactly a generic component - it actually heavily depends on the state and it has checks for atomic sites, domain only sites and etc. However I've fixed the `CurrentSite` component as it is a generic component.